app/src/main/java/org/mozilla/focus/fragment/UrlInputFragment.kt

```diff
@@ -712,32 +712,57 @@ class UrlInputFragment :
     private fun normalizeUrlAndSearchTerms(input: String): Triple<Boolean, String, String?> {
         val isUrl = UrlUtils.isUrl(input)
 
+        val shortCutAndQuery = UrlUtils.splitShortcutFromQuery(input)
+
         val url = if (isUrl)
             UrlUtils.normalize(input)
-        else
+        else if (shortCutAndQuery != null && getURLForSearchEngineShortcut(shortCutAndQuery[0], shortCutAndQuery[1]) != null) {
+            getURLForSearchEngineShortcut(shortCutAndQuery[0], shortCutAndQuery[1])!!
+        } else {
             UrlUtils.createSearchUrl(context, input)
+        }
 
         val searchTerms = if (isUrl)
             null
         else
             input.trim { it <= ' ' }
+
         return Triple(isUrl, url, searchTerms)
     }
 
     private fun onSearch(query: String, isSuggestion: Boolean = false, alwaysSearch: Boolean = false) {
-        if (alwaysSearch) {
-            openUrl(UrlUtils.createSearchUrl(context, query), query)
+        val shortCutAndQuery = UrlUtils.splitShortcutFromQuery(query)
```

Comment:
this should probably be handled by the viewmodel. We could create a search method on the `SearchSuggestionsViewModel` like `search` and move the logic there. Then create a new livedata that exposes the result of the search which this fragment can observe and handle

Reply:
This will also have the benefit of making it a lot easier to test since we can replace Android unit tests with jvm unit tests. The `SearchSuggestionsViewModelTest` test cases have good examples and we can add our test cases to the class

Reply:
Are you talking about moving only the logic related to search shortcuts within `onSearch` to `SearchSuggestionsViewModel`? Because if not, wouldn't that be out of scope for this PR?

Reply:
it depends tbh because these methods are logic that should technically be in the ViewModel instead of the view layer. It might have been the case that the focus developers didn't feel the need to test this part of the code and we end up being forced to refactor it a bit to accommodate our testing criteria. 

An alternative is to follow the pattern and not touch anything, but move the `getURLForSearchEngineShortcut` to the `UrlUtils` class and change it to accept a `SearchEngine` so it doesn't need to be aware of context

```
private fun getURLForSearchEngineShortcut(engine: SearchEngine, shortcut: String, actualQuery: String): String? {

        val currSearchEngine = UrlUtils.searchEngineShortcutsMap[shortcut]

        if (currSearchEngine != null) {
            return SearchEngine.buildSearchUrl(actualQuery)
        }

        return null
    }
```
and add a test to the `UrlUtilsTest` class.